### ISSUE TITLE
oneview_ethernet_network bug

### ISSUE THREAD

**Issue by rismoney** [HewlettPackard/terraform-provider-oneview#306]
So coincidentally I started to configure networks using terraform, and think I may have exposed the issue I was having related to network lookups happening improperly in profiles, but now I am seeing it on a much less convoluted resource for me implementation wise..   I am configuring about 25 vlan's using resources just like the ones below.
I snipped out the 20+ other vlan's to give a clean example.

```
resource "oneview_ethernet_network" "ETH_VLAN_150" {
  name    = "ETH_VLAN_150"
  type    = "ethernet-networkV4"
  vlan_id = 150
  smart_link = true
}
resource "oneview_ethernet_network" "ETH_VLAN_672" {
  name    = "ETH_VLAN_672"
  type    = "ethernet-networkV4"
  vlan_id = 672
  smart_link = true
}
```

No code change between subsequent TF plans:

```
PS C:\gitrepos\terraform\networks> C:\terraform12\terraform.exe plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

oneview_ethernet_network.ETH_VLAN_672: Refreshing state... [id=ETH_VLAN_672]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # oneview_ethernet_network.ETH_VLAN_150 must be replaced
-/+ resource "oneview_ethernet_network" "ETH_VLAN_150" {
      ~ category                = "ethernet-networks" -> (known after apply)
      ~ connection_template_uri = "/rest/connection-templates/2b7dc71e-2ff5-46cd-b004-5229f473942e" -> (known after apply)
      ~ created                 = "2021-04-13T15:35:37.822Z" -> (known after apply)
      ~ etag                    = "503ed9aa-70b0-4205-bd04-6105c6f29dd3" -> (known after apply)
        ethernet_network_type   = "Tagged"
      ~ fabric_uri              = "/rest/fabrics/60fd6f94-2d16-43d6-bbd7-90531bf45bda" -> (known after apply)
      ~ id                      = "ETH_VLAN_150" -> (known after apply)
      - initial_scope_uris      = [] -> null
      ~ modified                = "2021-04-13T15:35:37.822Z" -> (known after apply)
      ~ name                    = "ETH_VLAN_672" -> "ETH_VLAN_150"
        private_network         = false
        purpose                 = "General"
      ~ scopesuri               = "/rest/scopes/resources/rest/ethernet-networks/02ec89b4-2573-4a7f-a7a4-2ad3344c26b4" -> (known after apply)
        smart_link              = true
      ~ state                   = "Active" -> (known after apply)
      ~ status                  = "OK" -> (known after apply)
        type                    = "ethernet-networkV4"
      ~ uri                     = "/rest/ethernet-networks/02ec89b4-2573-4a7f-a7a4-2ad3344c26b4" -> (known after apply)
      ~ vlan_id                 = 672 -> 150 # forces replacement
    }

Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

5 seconds later I rerun TF:

```
PS C:\gitrepos\terraform\networks> C:\terraform12\terraform.exe plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

oneview_ethernet_network.ETH_VLAN_150: Refreshing state... [id=ETH_VLAN_150]
oneview_ethernet_network.ETH_VLAN_672: Refreshing state... [id=ETH_VLAN_672]


------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

**Comment by rismoney** [comment]
To be clear, If I re-run this multiple times, every time I plan I get varying results, where the networks are not properly correlating with the objects in oneview. It's seemingly random which networks don't matchup.

**Comment by rismoney** [comment]
One more point - TF was running on windows systems - not sure if that makes a difference, but one more data point.

[networks.txt](https://github.com/HewlettPackard/terraform-provider-oneview/files/6452442/networks.txt)

**Comment by rismoney** [comment]
This is really starting to become a major issue for us.  It's not just network resources, its the profiles as well.
There must be a bug when there are many items in the state file, and it's caching or resolving improperly.  Everytime I rerun a plan/apply I get different results.  

Let me know what I need to do to expedite.

**Comment by nabhajit-ray** [comment]
We are able to reproduce the issue. We are working on this and will get back to you.

**Comment by rismoney** [comment]
Appreciate that - this is probably the biggest bug in terms of priority, because it has now forced us to do this on profiles because idempotency is a hard requirement, when running TF in a CI/CD pipeline.

```
  lifecycle {
    ignore_changes = [
      name,
      boot,
      bios_option,
      hardware_name,
      connection_settings[0].connections[0],
      connection_settings[0].connections[1],
      connection_settings[0].connections[2],
      connection_settings[0].connections[3],
      connection_settings[0].connections[4],
      connection_settings[0].connections[5],
      local_storage[0].controller[0].logical_drives[0].bootable,
      local_storage[0].controller[0].logical_drives[0].name
    ]
  }
```

**Comment by nabhajit-ray** [comment]
We have found the root cause of this issue. The fix will be provided as part of OV 6.20

**Comment by nabhajit-ray** [comment]
The issue is already fixed and it will be available with OV 6.20 . The OV release is scheduled for mid July.

**Comment by AsisBagga** [comment]
@rismoney Hi Rich, 
This issue has been fixed in v6.2.0, and, therefore, I will be proceeding to close this issue. 
If you have any other concerns please feel free to open a new issue. Thank you.
